packages/custom-functions-metadata/src/parseTree.ts

```diff
@@ -1029,50 +1055,84 @@ function hasInvocationParameter(
-  if (t) {
-    let kind = t.kind;
-    const typePosition = getPosition(t);
-    if (ts.isTypeReferenceNode(t) || ts.isArrayTypeNode(t)) {
-      let arrayType: IArrayType = {
-        dimensionality: 0,
-        type: ts.SyntaxKind.AnyKeyword,
-      };
-      if (ts.isTypeReferenceNode(t)) {
-        const array = t as ts.TypeReferenceNode;
-        if (enumList.indexOf(array.typeName.getText()) >= 0) {
-          // Type found in the enumList
-          return type;
-        }
-        if (array.typeName.getText() !== "Array") {
-          extra.errors.push(logError("Invalid type: " + array.typeName.getText(), typePosition));
-          return type;
-        }
-      }
-      arrayType = getArrayDimensionalityAndType(t);
-      kind = arrayType.type;
+  // Only get type for typescript files. js files will return "any" for all types
+  if (!node) {
+    return type;
+  }
+  const typePosition = getPosition(node);
+  
+  // Get the inner type node if it is an array
+  if (typeNodeIsArray(node)) {
+    let arrayType: IArrayType = {
+      dimensionality: 0,
+      node: node,
+    };
+    arrayType = getArrayDimensionalityAndType(node);
+    node = arrayType.node;
+  }
+
+  // We currently accept the following types of reference node: enum will be converted to "any",
+  // Excel.CellValue will be converted accordingly. Anything else is invalid. (Array reference node has already been covered above.)
+  if (ts.isTypeReferenceNode(node)) {
+    const typeReferenceNode = node as ts.TypeReferenceNode;
+    let nodeTypeName = typeReferenceNode.typeName.getText();
+    if (enumList.indexOf(nodeTypeName) >= 0) {
+      // Type found in the enumList
+      return type;
     }
     // @ts-ignore
-    type = TYPE_MAPPINGS[kind];
-    if (!type) {
-      extra.errors.push(logError("Type doesn't match mappings", typePosition));
+    if (CELLVALUETYPE_MAPPINGS[nodeTypeName]) {
+      // @ts-ignore
+      let cellValue = CELLVALUETYPE_MAPPINGS[nodeTypeName]
+      if (cellValue === "unsupported") {
+        extra.errors.push(logError("Custom function does not support cell value type: " + typeReferenceNode.typeName.getText(), typePosition));
+        return type;
+      }
+      return cellValue;
     }
+
+    extra.errors.push(logError("Invalid type: " + typeReferenceNode.typeName.getText(), typePosition));
+    return type;
+  }
+
+  // @ts-ignore
+  type = TYPE_MAPPINGS[node.kind];
+  if (!type) {
+    extra.errors.push(logError("Type doesn't match mappings", typePosition));
   }
+ 
   return type;
 }
 
+/**
+ * Helper function that checks whether a TypeNode is an array.
+ * There are two cases: Array<sometype> or sometype[].
+ * @param node TypeNode
+ */
+function typeNodeIsArray(node: ts.TypeNode): boolean {
+  // [] case
+  if (ts.isArrayTypeNode(node)) {
+    return true;
+  }
+  // Array<sometype> case
+  if (ts.isTypeReferenceNode(node)) {
+    return (node as ts.TypeReferenceNode).typeName.getText() === "Array";
+  }
+  return false
```

Comment:
semicolon is missing at the end of the line.

Reply:
Make sure to run prettier if you're not having prettier applied in VS Code.

Reply:
Thanks. I ran prettier this iteration.